### PR TITLE
#0: CCL cleanup/fix

### PR DESCRIPTION
**Pull request by Aswinmcw**
### Ticket
No specific ticket

### Problem description
Needs some improvement and cleanup

### What's changed

- Fix `all_reduce` docs and README
- Replace if..else with switch case for better handling topologies
- Improve TT_FATAL error messages
- Cleanup unused imports

### Checklist
- [x] Post commit CI passes - https://github.com/tenstorrent/tt-metal/actions/runs/11436598174
- [x] T3K Pipelines - https://github.com/tenstorrent/tt-metal/actions/runs/11436601585
- [ ] Blackhole Post commit (if applicable)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [ ] New/Existing tests provide coverage for changes

### CODE REVIEW

File: ttnn/cpp/ttnn/operations/ccl/all_gather/device/all_gather_op.cpp [hunk 1]
```diff
@@ -136,7 +143,7 @@ AllGatherConfig::AllGatherConfig(Tensor const& input_tensor, Tensor const& outpu
 
 
 void AllGather::validate(const std::vector<Tensor> &input_tensors) const {
-    TT_FATAL(input_tensors.size() == 1, "Error");
+    TT_FATAL(input_tensors.size() == 1, "Error, Input tensor size should be 1");
```
Comment (SeanNijjar) [inline ttnn/cpp/ttnn/operations/ccl/all_gather/device/all_gather_op.cpp:146]: Can you please also include the size of the tensors list in the error message?

File: ttnn/cpp/ttnn/operations/ccl/all_gather/device/all_gather_op.cpp [hunk 2]
```diff
@@ -148,9 +155,9 @@ void AllGather::validate(const std::vector<Tensor> &input_tensors) const {
     // TODO: Validate ring
     TT_FATAL(input_tensor.storage_type() == StorageType::DEVICE, "Operands to all_gather need to be on device!");
     TT_FATAL(input_tensor.buffer() != nullptr , "Operands to all_gather need to be allocated in buffers on device!");
-    TT_FATAL(this->num_links > 0, "Error");
+    TT_FATAL(this->num_links > 0, "Error, num_links should be more than 0");
     TT_FATAL(this->num_links <= input_tensor.device()->compute_with_storage_grid_size().y, "Worker cores used by links are parallelizaed over rows");
-    TT_FATAL(this->receiver_device_id.has_value() || this->sender_device_id.has_value(), "Error");
+    TT_FATAL(this->receiver_device_id.has_value() || this->sender_device_id.has_value(), "Error, Either receiver or sender device needs to have some value");
```
Comment (SeanNijjar) [inline ttnn/cpp/ttnn/operations/ccl/all_gather/device/all_gather_op.cpp:160]: Can we try something more informative such as All-gather was unable to identify either a sender or receiver device ID and atleast one must be identified for a valid all-gather configuration. The input mesh tensor or all-gather arguments may be incorrect"

File: ttnn/cpp/ttnn/operations/ccl/reduce_scatter/device/reduce_scatter_op.cpp [hunk 3]
```diff
@@ -110,7 +110,7 @@ Tensor reduce_scatter(
                     break;
                 }
             }
-            TT_FATAL(receiver_device_id != std::nullopt || sender_device_id != std::nullopt, "Error in reduce scatter op setup");
+            TT_FATAL(receiver_device_id != std::nullopt || sender_device_id != std::nullopt, "Error in reduce scatter op setup, Either receiver or sender device needs to have some value");
```
Comment (SeanNijjar) [inline ttnn/cpp/ttnn/operations/ccl/reduce_scatter/device/reduce_scatter_op.cpp:113]: Same comment as above.

**Review by SeanNijjar — APPROVED** [review]
Thanks a lot for taking a look at this cleanup unprompted :)

I left some suggestions for minor changes to the error message.

**Review by jdesousa-TT — APPROVED** [review]
Please run these ccl tests on the sweeps pipeline to check for import errors. It doesn't have to pass but if the generation finishes then we know theres no import issues.

**Review by ayerofieiev-tt — APPROVED** [review]

**Pull request merged**